File: backend/src/services/emailFinder.js
// Lightweight email discovery stub.
// Replace with real search / people-API integration (SerpAPI, Hunter, Clearbit, etc.).

// Parse artist input (comma/newline separated) and log when called.
function parseArtists(artistsStr) {
  const list = (artistsStr || '')
    .split(/[\n,]+/)
    .map(s => s.trim())
    .filter(Boolean)

  console.log('artists done parsing:', list.length, list)
  return list
}

// Find emails for given artist names and log discovery counts when called.
async function findEmailsForArtists(artists) {
  const out = {}
  for (const a of artists) {
    const name = a.trim()
    if (!name) continue
    // create a naive normalized handle for demo purposes
    const norm = name.toLowerCase().replace(/\s+/g, '.').replace(/[^a-z0-9.]/g, '')
    out[name] = [
      `${norm}@example.com`,
      `contact@${norm}music.com`
    ]
  }

  // count discovered emails and log
  let total = 0
  for (const k of Object.keys(out)) total += (out[k] || []).length
  console.log('found', total, 'emails')

  // Log each discovered email per artist for visibility
  for (const artist of Object.keys(out)) {
    console.log('discovered for', artist + ':', out[artist])
  }

  return out
}

module.exports = { parseArtists, findEmailsForArtists };

File: backend/app/services/email_finder.py
from __future__ import annotations

import re
from typing import Any


def parse_artists(artists_str: str | None) -> list[str]:
    artists = (artists_str or "").split(r"[\n,]+") if False else re.split(r"[\n,]+", artists_str or "")
    parsed = [artist.strip() for artist in artists if artist and artist.strip()]
    print(f"artists done parsing: {len(parsed)} {parsed}")
    return parsed


async def find_emails_for_artists(artists: list[str]) -> dict[str, list[str]]:
    out: dict[str, list[str]] = {}

    for artist in artists:
        name = artist.strip()
        if not name:
            continue
        # For now, generate placeholder emails but log a warning
        # TODO: Extract real emails from research results
        normalized = name.lower().replace(" ", ".").replace("-", ".")
        normalized = re.sub(r"[^a-z0-9.]", "", normalized)
        emails = [f"{normalized}@example.com", f"contact@{normalized}music.com"]
        out[name] = emails
        print(f"[EMAIL_FINDER] {name}: Generated placeholder emails (should extract from research results)")
        for email in emails:
            print(f"  - {email}")

    total = sum(len(emails) for emails in out.values())
    print(f"[EMAIL_FINDER] Total placeholder emails: {total}")
    print(f"[EMAIL_FINDER] ⚠️  WARNING: Using placeholder emails - implement extraction from research results!")

    return out
